### PR TITLE
Copy Blob from URL/Put Blob from URL copy source tags 

### PR DESCRIPTION
**Pull request by siminsavani-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/blockblob/models.go [hunk 1]
```diff
@@ -23,6 +23,14 @@ type Block = generated.Block
 // BlockList - can be uncommitted or committed blocks (committed/uncommitted)
 type BlockList = generated.BlockList
 
+// BlobCopySourceTags - can be 'COPY' or 'REPLACE'
+type BlobCopySourceTags = generated.BlobCopySourceTags
+
+const (
+	BlobCopySourceTagsCOPY    BlobCopySourceTags = "COPY"
```
Comment (gapra-msft) [inline sdk/storage/azblob/blockblob/models.go:30]: should this be capitalized in the variable name?

File: sdk/storage/azblob/blockblob/client_test.go [hunk 2]
```diff
@@ -646,6 +646,83 @@ func (s *BlockBlobUnrecordedTestsSuite) TestPutBlobFromURL() {
 	_require.Equal(destBuffer, sourceData)
 }
 
+func (s *BlockBlobUnrecordedTestsSuite) TestPutBlobFromURLWithCopySourceTagsReplace() {
```
Comment (seanmcc-msft) [inline sdk/storage/azblob/blockblob/client_test.go:649]: Let's add a test case for not specifying x-ms-copy-source-tags-options, the default should be replace.

File: sdk/storage/azblob/blockblob/models.go (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/azblob/blockblob/models.go:30]: Yes, this is by-design.  There was a bug in the service requiring the enum values to be all caps.

[OneNote](https://microsoft.sharepoint.com/teams/AzureStorage/_layouts/OneNote.aspx?id=%2Fteams%2FAzureStorage%2FXStore%20Private%20OneNote%2FSTG%20Releases&wd=target%28STG%2073-87%2FSTG%2082.one%7C056FD56A-EDDB-4859-8C40-8D53D761DEC0%2FCopy%20Blob%20from%20URL%5C%2FPut%20Blob%20from%20URL%20copy%20source%7C6CEF33E7-28BA-4F88-8366-4F75A1D1B773%2F%29
onenote:https://microsoft.sharepoint.com/teams/AzureStorage/XStore%20Private%20OneNote/STG%20Releases/STG%2073-87/STG%2082.one#Copy%20Blob%20from%20URL/Put%20Blob%20from%20URL%20copy%20source&section-id={056FD56A-EDDB-4859-8C40-8D53D761DEC0}&page-id={6CEF33E7-28BA-4F88-8366-4F75A1D1B773}&end)

Comment (gapra-msft) [reply sdk/storage/azblob/blockblob/models.go:30]: yeah but there's no reason we need to be exposing variable names that aren't camel case right?

Comment (seanmcc-msft) [reply sdk/storage/azblob/blockblob/models.go:30]: Ya we should probably make the public-facing enum values camel case.

Comment (siminsavani-msft) [reply sdk/storage/azblob/blockblob/models.go]: Done! I made the variable names camel cased.

File: sdk/storage/azblob/blockblob/client_test.go (hunk 2 above)
Comment (siminsavani-msft) [reply sdk/storage/azblob/blockblob/client_test.go:687]: Added!

**Review by gapra-msft — APPROVED** [review]

File: sdk/storage/azblob/blockblob/models.go [hunk 3]
```diff
@@ -23,6 +23,14 @@ type Block = generated.Block
 // BlockList - can be uncommitted or committed blocks (committed/uncommitted)
 type BlockList = generated.BlockList
 
+// BlobCopySourceTags - can be 'COPY' or 'REPLACE'
+type BlobCopySourceTags = generated.BlobCopySourceTags
+
+const (
+	BlobCopySourceTagsCopy    BlobCopySourceTags = "COPY"
+	BlobCopySourceTagsReplace BlobCopySourceTags = "REPLACE"
+)
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/blockblob/models.go:32]: This constant type should be added in blockblob/constants.go.
 Also, the constants should be using the value from the generated code like,
```
BlobCopySourceTagsCopy    BlobCopySourceTags = generated.BlobCopySourceTagsCOPY
```

**Review by souravgupta-msft — APPROVED** [review]

File: sdk/storage/azblob/blockblob/models.go (hunk 3 above)
Comment (siminsavani-msft) [reply sdk/storage/azblob/blockblob/models.go:32]: Done!

File: sdk/storage/azblob/blockblob/models.go [hunk 4]
```diff
@@ -23,6 +23,14 @@ type Block = generated.Block
 // BlockList - can be uncommitted or committed blocks (committed/uncommitted)
 type BlockList = generated.BlockList
 
+// BlobCopySourceTags - can be 'COPY' or 'REPLACE'
+type BlobCopySourceTags = generated.BlobCopySourceTags
+
+const (
+	BlobCopySourceTagsCopy    = generated.BlobCopySourceTagsCOPY
+	BlobCopySourceTagsReplace = generated.BlobCopySourceTagsREPLACE
+)
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/blockblob/models.go:32]: This part should also be moved to `blockblob/contants.go`.

Comment (siminsavani-msft) [reply sdk/storage/azblob/blockblob/models.go]: Done!

**Pull request merged**